docs/source/enterprise-guide/change-management.mdx

```diff
@@ -0,0 +1,537 @@
+   so the release should have no impact on current behavior.
+3. Progressively enable the feature flag for more users, monitoring error rates
+   or support inquiries.
+4. Once the feature flag is enabled for 100% of users, remove the old codepath
+   and perform a clean-up release.
+
+**Canary deploys**: instead of adding complexity to our resolver by supporting
+two codepaths and selecting one based on a feature flag, we can choose to simply
+author the new resolver and release it more carefully.
+
+1. Author the new resolver.
+2. Release the new code using a canary deployment mechanism.
+3. Observe error rates or other relevant metrics (either manually, or with an
+   automated progressive delivery controller like [Flagger][flagger]).
+4. If metrics are acceptable, release the new code to all nodes.
+5. If metrics are unacceptable, cancel the canary deployment, roll back the
+   release, and revert the code change.
+
+[flagger]: https://flagger.app/
+
+## Changes to gateway logic or configuration
+
+The gateway is critical infrastructure and should be treated as such. Although
+we can treat releases as standard rolling deploys, it's a very good idea to use
+progressive delivery techniques like canary or blue-green deploys.
+
+### Progressive delivery techniques
+
+Deploying gateway changes without schema changes using a blue-green deployment
+process is straightforward, whether we're manually deploying to separate node
+pools or clusters, or using a automated progressive delivery controller like
+[Flagger](flagger).
+
+If we're also using
+[out-of-band schema publishing](#release-alternative-out-of-band-schema-publishing)
+to make shared type or query planning changes, we might choose to have a third
+pool or cluster of gateway instances configured to use a separate variant. This
+allows a progressive delivery controller to automatically manage the blue-green
+rollout of two pools/clusters configured to use the same variant, while
+reserving the third pool/cluster for out-of-band changes.
+
+## Summary
+
+It can feel overwhelming to consider all the ways the many components interact
+in a federated architecture. By considering the finite set of change scenarios
+independently, it becomes clearer which processes and techniques are useful.
+
+Key takeaways:
+
+- Isolate changes as much as possible — when possible, don't release schema
+  changes and implementation changes at the same time.
+- Sequencing gateway/subgraph schema changes isn't as critical in practice as we
+  might think — especially if we use tools like deprecation workflows, usage
+  reporting, and schema change notifications.
+- Blue-green deployment infrastructure for gateways is a great idea for safe
+  changes to high-uptime services.
+- Publish subgraph schemas to the "current" variant immediately and run build
+  checks against the "current" variant to catch subgraph compatibility issues
+  long before a production release.
```

Comment:
See my comments above, but I think I'm missing something about how you guys are thinking about a `Production` variant vs. a `current` variant.